protocol/x/subaccounts/keeper/transfer.go

```diff
@@ -273,6 +249,34 @@ func (k Keeper) DistributeFees(
 		); err != nil {
 			return err
 		}
+
+		// emit a metric for the amount of fees transferred to the market mapper
+		if revShare.RevShareType == revsharetypes.REV_SHARE_TYPE_MARKET_MAPPER {
+			labels := []metrics.Label{
+				metrics.GetLabelForIntValue(metrics.MarketId, int(perpetual.Params.MarketId)),
+			}
+			metrics.AddSampleWithLabels(
+				metrics.MarketMapperRevenueDistribution,
+				metrics.GetMetricValueFromBigInt(revShare.QuoteQuantums),
+				labels...,
+			)
+		}
+	}
+
+	totalRevShareQuoteQuantums := big.NewInt(0).Add(
+		revSharesForFill.FeeSourceToQuoteQuantums[revsharetypes.REV_SHARE_FEE_SOURCE_TAKER_FEE],
```

Comment:
The keys `revsharetypes.REV_SHARE_FEE_SOURCE_TAKER_FEE` and `revsharetypes.REV_SHARE_FEE_SOURCE_NET_FEE` aren't guaranteed to exist right?

Reply:
Since `revSharesForFill` are gotten from `GetAllRevShares` - the only way these keys arent set are if there is an error. Otherwise we know they exist since [I explicitly set them](https://github.com/dydxprotocol/v4-chain/pull/2251/files#diff-1203e511f1b1da0901525ce89b0abb29818a2cc88b7f5e916f0734df1a7d9de6R157-R160)

Reply:
But frankly, better to check either way - so will update!